docs/trace/extending-the-sdk/README.md

```diff
@@ -211,6 +211,12 @@ Writing an instrumentation library typically involves 3 steps.
          > Changing the name of the source should be considered a
          breaking change.
 
+4. The fourth step is an optional step, and involves making any member required
```

Comment:
isn't this covered in 3 already? The extension method should be public.

Reply:
clarified in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4347/commits/1b373781d7bd475cbe9e03d313281ac2f041d802

Reply:
@cijothomas The extension method is used during "boostrapping". We ask here for API that allows adding things to instrument at runtime (e.g. when a Redis connection is created).

Reply:
Instead of making it a 4th step, this could be a warning in the 3rd step: if enabling the instrumentation requires references to actual objects from the targeted library the instrumentation library should offer public APIs to pass such objects and not rely on the extension methods to receive such objects.

The problem of passing such objects via `AddInstrumentation` is that to correctly enable the instrumentation code that perhaps is encapsulated somewhere else now needs to be brought to the same place of the tracer initialization.

Reply:
replaced with warning in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4347/commits/9ff7ab0d23fbb22ab2e44ee1da2a548908c23b38